src/schema/v2/artwork/updateArtworkMutation.ts

```diff
@@ -0,0 +1,49 @@
+import { GraphQLString, GraphQLNonNull } from "graphql"
+import { mutationWithClientMutationId } from "graphql-relay"
+import { ResolverContext } from "types/graphql"
+import Artwork from "schema/v2/artwork"
+import { omitBy } from "lodash"
+
+interface UpdateArtworkMutationInputProps {
+  id: string
+  availability?: boolean
+}
+
+export const updateArtworkMutation = mutationWithClientMutationId<
+  UpdateArtworkMutationInputProps,
+  any,
+  ResolverContext
+>({
+  name: "UpdateArtworkMutation",
+  description: "Updates an artwork.",
+  inputFields: {
+    availability: {
+      type: GraphQLString,
+      description: "The availability of the artwork",
+    },
+    id: {
+      type: new GraphQLNonNull(GraphQLString),
+      description: "The id of the artwork to update.",
+    },
+  },
+  outputFields: {
+    artwork: {
+      type: Artwork.type,
+      resolve: (artwork) => artwork,
+    },
+  },
+  mutateAndGetPayload: async ({ id, ...props }, { updateArtworkLoader }) => {
+    if (!updateArtworkLoader) {
+      return new Error("You need to be signed in to perform this action")
+    }
+
+    const payload = omitBy(props, (prop) => prop == null)
```

Comment:
What's this for? (does our `fetch` library already leave out blanks, or b/c this is a mutation you want to be more careful to unset blank fields)? This raises the question, which I've found often does _not_ work at Artsy, cool so this lets you 'set' something, but what about 'unsetting' it?

Reply:
Yeah this just omits any null or undefined payloads from the object just to be safe. Eg:

```
{foo: undefined, bar: undefined, baz: "bam" } -> { baz: "bam }
```